test/telemetry_influx_db_test.exs

```diff
@@ -423,9 +592,49 @@ defmodule TelemetryInfluxDBTest do
     assert_tags(config, tags)
   end
 
+  defp do_assert_reported(%{version: :v2} = config, name, values, tags) do
+    results =
+      eventually(fn ->
+        res = query(config, name)
+
+        if empty_result?(config, res) do
+          false
+        else
+          res
+        end
+      end)
+
+    tag_values = Map.values(tags)
+    tag_keys = Map.keys(tags)
+
+    assert Enum.all?(results, fn result ->
+             measurement = Map.get(result, "_measurement")
+
+             has_tag_keys = Enum.all?(tag_keys, fn tag_key -> Map.has_key?(result, tag_key) end)
+
+             result_tag_values =
+               tags
+               |> Map.keys()
+               |> Enum.map(fn key -> Map.get(result, key) end)
+
+             measurement == name and has_tag_keys and result_tag_values == tag_values
+           end)
+
+    assert Enum.map(results, fn result -> Map.get(result, "_field") end) == Map.keys(values)
+
+    Enum.each(Map.keys(values), fn key ->
```

Comment:
I was thinking about decomposing it in some way like:
``` elixir
    values
    |> Map.keys()
    |> Enum.each(&assert_values(&1, results, values))
  end

  defp assert_value(key, results, values) do
    field_result = Enum.find(results, & &1["_field"] == key)
    assert field_result["_value"] == values[key]
  end
```
not sure if `assert_value` would be the best name here... 
What do you think ?